src/or/dirserv.c

```diff
@@ -2673,15 +2676,22 @@ dirserv_read_measured_bandwidths(const char *from_file,
         dirserv_cache_measured_bw(&parsed_line, file_time);
         if (measured_bw_line_apply(&parsed_line, routerstatuses) > 0)
           applied_lines++;
-      } else {
-        /* If line does not contain the header separator and it is key_value,
-         * it is probably a KeyValue header.*/
-        if (strcmp(line, "====\n") != 0 &&
-            string_is_key_value(LOG_DEBUG, line)) {
-          line[strlen(line)-1] = '\0';
-          smartlist_add_strdup(bwlist_headers, line);
-        };
-      }
+      /* if the terminator is found, it is the end of header lines, set the
+       * flag but do not store anything */
+      } else if (strcmp(line, BW_FILE_TERMINATOR) == 0)
+        line_is_after_headers = 1;
+      /* if the line was not a correct relay line nor the terminator and
+       * the end of the header lines has not been detected yet
+       * and it is key_value and bw_file_headers did not reach the maximum
+       * number of headers,
+       * then assume this line is a header and add it to bw_file_headers */
+      else if (bw_file_headers &&
+              (line_is_after_headers == 0) &&
+              string_is_key_value(LOG_DEBUG, line) &&
```

Comment:
string_is_key_value returns true for relay bandwidth lines, because it allows spaces:
https://github.com/torproject/tor/blob/be40ad51b68fa87bafb88e55db7856d373274a4d/src/lib/encoding/keyval.c#L19

Please also check that there are no spaces in the line.